src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala

```diff
@@ -268,4 +272,259 @@ trait VLQReaderWriterSpecification extends PropSpec
+  }
+
+  property("unsigned Short roundtrip") {
+    forAll { v: Short =>
+      val uv: Int = v & 0xFFFF
+      byteBufReader(byteArrayWriter().putUShort(uv).toBytes).getUShort() shouldBe uv
+    }
+  }
+
+  property("Int roundtrip") {
+    forAll { v: Int => byteBufReader(byteArrayWriter().putInt(v).toBytes).getInt() shouldBe v }
+  }
+
+  property("unsigned Int roundtrip") {
+    forAll { v: Int =>
+      val uv: Long = v.toLong + (Int.MinValue.toLong * -1)
+      byteBufReader(byteArrayWriter().putUInt(uv).toBytes).getUInt() shouldBe uv
+    }
+  }
+
+  property("Long roundtrip") {
+    forAll { v: Long => byteBufReader(byteArrayWriter().putLong(v).toBytes).getLong() shouldBe v }
+  }
+
+  property("ULong roundtrip") {
+    forAll { v: Long => byteBufReader(byteArrayWriter().putULong(v).toBytes).getULong() shouldBe v }
+  }
+
+  property("Boolean array roundtrip") {
+    forAll { v: Array[Boolean] => byteBufReader(byteArrayWriter().putBits(v).toBytes).getBits(v.length) shouldBe v }
+  }
+
+  property("short string roundtrip") {
+    forAll(Gen.alphaStr.suchThat(_.length < 256)) { v: String =>
+      byteBufReader(byteArrayWriter().putShortString(v).toBytes).getShortString() shouldBe v
+    }
+  }
+
+  property("byte corner cases") {
+    def roundtrip(v: Byte, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().put(v).toBytes
+      bytes shouldEqual expected
+      byteBufReader(expected).getByte() shouldBe v
+    }
+
+    roundtrip(Byte.MinValue, bytesFromInts(Byte.MinValue))
+    roundtrip(0, bytesFromInts(0))
+    roundtrip(Byte.MaxValue, bytesFromInts(Byte.MaxValue))
+  }
+
+  property("unsigned byte corner cases") {
+    def roundtrip(v: Int, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().putUByte(v).toBytes
+      bytes shouldEqual expected
+      byteBufReader(expected).getUByte() shouldBe v
+    }
+
+    an[IllegalArgumentException] should be thrownBy roundtrip(-1, bytesFromInts(0))
+    roundtrip(0, bytesFromInts(0))
+    roundtrip(255, bytesFromInts(255))
+    an[IllegalArgumentException] should be thrownBy roundtrip(256, bytesFromInts(0))
+  }
+
+  private def prettyPrint(arr: Array[Byte]): String =
+    arr.map(b => String.format("0x%02X", Byte.box(b))).mkString(", ")
+
+  property("Short corner cases") {
+    def roundtrip(v: Short, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().putShort(v).toBytes
+      withClue(s"for value $v got bytes ${prettyPrint(bytes)} (expected ${prettyPrint(expected)}): \n") {
+        bytes shouldEqual expected
+      }
+      byteBufReader(expected).getShort() shouldBe v
+    }
+
+    roundtrip(Short.MinValue, bytesFromInts(0xFF, 0xFF, 0x03))
+    roundtrip(-8193, bytesFromInts(0x81, 0x80, 0x01))
+    roundtrip(-8192, bytesFromInts(0xFF, 0x7F))
+    roundtrip(-65, bytesFromInts(0x81, 0x01))
+    roundtrip(-64, bytesFromInts(0x7F))
+    roundtrip(0, bytesFromInts(0))
```

Comment:
suggest to add the following cases: -8194, -8191, -66, -63, -1, 1, 62, 65, 8190, 8193
Also needed test cases for which show what happens when (0x81, 0x00) is read (and the same for all other whole in the byte arrays space)

Reply:
Done.